artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/ActiveMQDestination.java

```diff
@@ -309,7 +330,7 @@ public SimpleString getSimpleAddress() {
    }
 
    public String getName() {
-      return simpleAddress.toString();
+      return name == null ? simpleAddress.toString() : name.toString();
```

Comment:
if we use `address` instead of `null` when creating an ActiveMQDestination, we ensure that the fields are always non-null and this method can just return `name.toString()`

Reply:
Fair enough.